Scenarios/StorageMigrationService/scenario.ps1

```diff
@@ -0,0 +1,81 @@
+﻿###############
+# Run from DC #
+###############
+
+# Verify Running as Admin
+$isAdmin = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
+If (!( $isAdmin )) {
+    Write-Host "-- Restarting as Administrator" -ForegroundColor Cyan ; Start-Sleep -Seconds 1
+    Start-Process powershell.exe "-NoProfile -ExecutionPolicy Bypass -File `"$PSCommandPath`"" -Verb RunAs 
+    exit
+}
+
+$StartDateTime = get-date
+Write-host "Scripts started at $StartDateTime"
+
+#Set PSScriptRoot
+$PSScriptRootFolder = "D:\Scripts"
+
+##### LAB Config #####
+#Change the Servers here based on the servers you need to deploy
+
+$WAC = 'WAC'
+$SMS_2008R2 = 'SMS2008R2'
+$SMS_2012R2 = 'SMS_2012R2'
+$SMS_2019 = 'SMS_2019'
+
+$Servers = ($WAC, $SMS_2012R2, $SMS_2019)
+
+#Enable firewall rules for servers
+Invoke-Command -ComputerName $Servers -ScriptBlock { Get-NetFirewallRule -Name *FPS* | Enable-NetFirewallRule ; Enable-PSRemoting -Force -Confirm:$false }
+Invoke-Command -ComputerName $Servers -ScriptBlock { Get-NetFirewallRule -Name *rpc* | Enable-NetFirewallRule }
+Invoke-Command -ComputerName $Servers -ScriptBlock { Get-NetFirewallRule -Name *wmi* | Enable-NetFirewallRule } 
+
+#Upgrade Powershell on 2008R2 server
+IF ($SMS_2008R2) {    
+    #Install requierd Roles on 2008R2 Server
+    Write-host "Installing IIS on 2008R2 Server"
+    Invoke-Command -ComputerName $SMS_2008R2 -ScriptBlock {
+        Import-Module Servermanager
+        Add-WindowsFeature Web-server -IncludeAllSubFeature
+    }
+
+    #Wait for servers to come online before resuming
+    Write-host "Waiting for SMS2008R2 to come online again"
+    Restart-Computer -ComputerName $SMS_2008R2 -Protocol wsman -Wait -Force
+    Start-Sleep -Seconds 30
+
+    ##Copy IIS files Hello World
+    Write-host "Copy IIS files to SMS2008R2"
+    Copy-Item "$PSScriptRootFolder\iisstart.htm" -Destination "\\$SMS_2008R2\c$\inetpub\wwwroot"
```

Comment:
SMB might not be enabled on firewall. I'll rewrite it to copy using PSSession if you don't mind.

Reply:
It works :)